src/cairoUtilFuncGen/serialisation.ts

```diff
@@ -41,13 +62,23 @@ export function serialiseReads(
 enum Read {
   Felt,
   Id,
+  Address,
+  UN,
   Bool,
 }
 
-function producePackExpression(type: CairoType): (string | Read)[] {
-  if (type instanceof WarpLocation) return [Read.Id];
-  if (type instanceof CairoFelt) return [Read.Felt];
-  if (type instanceof CairoBool) return [Read.Bool];
+interface packExpReturnType {
+  dataOrDataType: string | Read;
+  metadata: {
+    nBits?: number;
+  };
+}
+
+function producePackExpression(type: CairoType): packExpReturnType[] {
+  if (type instanceof WarpLocation) return [{ dataOrDataType: Read.Id, metadata: {} }];
+  if (type instanceof CairoFelt) return [{ dataOrDataType: Read.Felt, metadata: {} }];
+  if (type instanceof CairoContractAddress) return [{ dataOrDataType: Read.Address, metadata: {} }];
+  if (type instanceof CairoBool) return [{ dataOrDataType: Read.Bool, metadata: {} }];
```

Comment:
Imo it'd be more readable to create a function wrapping `Read.<type>` within the `packExpReturnType[]` instead of repeating it in each lane

Reply:
Similarly the code below